### DISCUSSION TITLE
Improve navigation for linked hashes/hash management

### DISCUSSION THREAD

**Discussion by televandalist** [RetroAchievements/RAWeb#982] [Ideas]
Some suggestions:

1. Upon creating an official forum topic for an entry, automatically add a link to an entry's linkedhashes.php page in the first post.

2. For linkedhashes.php, add a link to that entry's managehashes.php page. 

3. For managehashes.php, add a link to that entry's forum topic and linkedhashes.php page.

**Comment by MrOwnership** [comment]
I think 1 would be better off implemented by updating the forum topic page to display the hashes programmatically such that only forum topics that are linked to a game will have this implementation. Then we can get rid of all the hash information in the first posts, those are a pain to upkeep now that we don't need to. Two ways that came to mind are:

List the hash info at the top of the page:
![image](https://user-images.githubusercontent.com/16140035/168484101-5678d891-d568-4ebe-995a-dd1bef05b4ee.png)

Put a link to the linked hashes page at the top of the page:
![image](https://user-images.githubusercontent.com/16140035/168484237-0c05bf92-9458-4eae-9267-9332650dd545.png)

The second option being the cleaner and easier implementation, but doesn't immediately display as much information as the first. In both of these cases the information would show at the top of the forum topic regardless of which page the user is on, making it so they don't need to go to the first page to find the hash information.

2 and 3 are simple enough to do.

**Comment by Jamiras** [comment]
I don't like either of these solutions. The hash list is already linked directly from the game page right next to the link for the official forum topic. Merging it into the official forum topic was a bandage before it had its own dedicated page.

![image](https://user-images.githubusercontent.com/32680403/168489986-16e6733c-2aa0-492f-bb9b-647c5a40601f.png)

**Comment by Blazekickn** [comment]
i usually check the forum topic over the linked hashes page because on the forum topic you can provide info on the set and link to patches or homebrew. the current proof of concept doesn't look great but hash info should be listed in the forum topic somewhere

**Comment by televandalist** [comment]
> because on the forum topic you can provide info on the set and link to patches or homebrew

I'd prefer to just beef up the linkedhashes.php page so we can make it completely unnecessary to use the forum for hash info. The forum posts require way too much maintenance and the info is too important to just have "loose" in the forum. 

> The hash list is already linked directly from the game page right next to the link for the official forum topic.

What I'm thinking is to have the links to the hashes, code notes, tickets, leaderboards, etc. at the start of the forum topic, similar to how MrO shows the link in his second implementation. I'll see if I can whip up some examples cause I'll end up just embarrassing myself with the terminology 😆

**Comment by Jamiras** [comment]
My point is that the "official forum topic" is still _just a forum topic_. You can get to them through the Community > Forums page. If we really wanted to do a one-off page like this, I'd recommend separating them from the main forums entirely and only have them accessible from the game page. However, that removes the ability to generally browse new forum posts to see which official forum topics are active - which goes back to the point that there are just forum topics specifically linked to a game.

I'm fine with embedding the links in the first post (as a replacement or addendum to putting the hashes directly in there). If we put them outside of the posts, do they still show up when you go to the Next Page?

Maybe we can make a special container page for the official forum topic that has a forum page on the left side and the existing sidebar (links, related games/categories, maybe even leaderboards) on the right.

**Comment by luchaos** [comment]
A forum / topic linked to a game can have basic game infos linked in the sidebar from where it's possible to drill down into linked hashes/patches and other information. The "single topic per game" paradigm will fall.
Think discussions on Steam. Requirements are possibilities to tag, lock, pin topics.
